pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -124,10 +131,15 @@ public class TableRebalancer {
   private final ControllerMetrics _controllerMetrics;
   private final RebalancePreChecker _rebalancePreChecker;
   private final TableSizeReader _tableSizeReader;
+  private final ExecutorService _executorService;
+  private final HttpClientConnectionManager _connectionManager;
```

Comment:
let's not pass this in if it can be avoided

In the `DefaultPreChecker` we create a connection manager. I would recommend doing the same here:
```
    try (PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager()) {
 ...
 ...
    }
```

Also recommend using a try-with-resources type block as mentioned above so that the connection manager is automatically closed when no longer needed